.github/workflows/integration-tests-non-root.yaml

```diff
@@ -70,6 +70,48 @@ jobs:
           chown -Rf ci:ci ${GITHUB_WORKSPACE} $(go env GOMODCACHE) $(go env GOCACHE)
         continue-on-error: true
 
+      - name: Install Docker CLI
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+              ca-certificates \
+              curl \
+              gnupg
+          sudo install -m 0755 -d /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          sudo chmod a+r /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+            "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+            sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt-get update
+          sudo apt-get install -y docker-ce-cli
+
+      - name: Create KinD cluster
```

Comment:
How does this affect the job runtime? Would it make sense to move integration Kube tests into a separate workflow instead?

Reply:
Creating the cluster takes ~40s and running the kube integration tests takes more or less 110s

Reply:
Will do it monday

Reply:
Done but we have a problem. 
Tests fail from time to time because of a data race on a repo used by Kubernetes tooling. 
I already created the PR with a fix: https://github.com/moby/spdystream/pull/91
This affects `TestExec` and `TestJoin`